modules/server/src/main/ts/bedrock/util/Arr.ts

```diff
@@ -8,3 +8,14 @@ export const bind = <A, B> (xs: A[], f: (t: A) => B[]): B[] =>
 
 export const bind2 = <A, B, C> (xs: A[], f: (a: A) => B[], g: (b: B) => C[]): C[] =>
   bind(bind(xs, f), g);
+
+export const findMap = <A, B> (xs: ArrayLike<A>, f: (a: A, index: number) => B | null): B | null => {
+  for (let i = 0; i < xs.length; i++) {
+    const result = f(xs[i], i);
```

Comment:
Are we good to leave this untested? It's super basic, so it should be fine right?

Reply:
It's basically a direct copy from `Arr` in the monorepo, just without `Optional` since we can't use that here. It might be worth adding a test though :shrug:

Reply:
As discussed we probably should, but it doesn't have tests atm so not worth the effort atm. Ideally, we should probably look at using katamari, since bedrock-server wouldn't have a circular dependency and therefore could actually use it.